core/server/data/migrations/versions/3.25/02-add-amp-gtag-id-setting.js

```diff
@@ -0,0 +1,20 @@
+const logging = require('../../../../../shared/logging');
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up({transacting: knex}) {
+        logging.info('Updating amp_gtag_id setting to amp group');
+        await knex('settings')
+            .update({
+                group: 'amp'
+            })
+            .where({
+                key: 'amp_gtag_id'
+            });
+    },
+
+    async down() {}
```

Comment:
Ok probably another Q which I've missed some context on, but why aren't we removing `amp_gtag_id` field completely when rolling back? Might be an edge case that we reuse the `key` name through different Ghost versions but seems rational to me to leave the DB in the same state - without a new field when rolling back :thinking:

Reply:
There has been no discussion on it as of yet and is probably a good thing to raise, but since we never had any explicit migrations for new settings so far, we also never had any `down` migration removing any new setting on rollback. All new settings so far have stayed in the DB inspite the of the rollback, which is something that probably needs bigger discussion on how to handle as definitely doesn't sound ideal ?

Reply:
@ErisDS summary:

The setting is not added by the migration explicitly, but the `group` and `tags` fields will be set in an explicit migration in the version in which the setting is added

1. Should we add the setting (or [update _all_ fields](https://github.com/TryGhost/Ghost/pull/12039#discussion_r453548690)) as part of the explicit migration?
2. Should the rollback completely remove the setting?

imo it would be good to establish this convention early now that we require explicit migrations for each new setting.

Reply:
Am still thinking on this - will get back to you later today

Reply:
I think we should stick to doing what is needed and expected for this particular migration. It's just adding missing column info, so the down migration should be equivalent.

If we do anything else, we pre-empting what these migrations need to look like in a world where the underlying problems are solved, and realistically until we know what that is actually going to look like we can't do that without making a bunch of dangerous assumptions. So we're likely creating more work that would need to be redone.

E.g. if we get rid of populateDefaults, then we _may_ have to rewrite a lot of these migrations in future, but really trying to figure that out requires solving the problem and we/I don't have time for that right now.

If anyone disagrees, let me know - but I think for now KISS is best.

Reply:
> It's just adding missing column info, so the down migration should be equivalent.

There is a problem with this because we cannot be certain what the `group` column was set to before the migration, unless we start reading backup files (out of scope of this issue & appetite)

I would suggest that because this migration is purely to correct the state of the database that it doesn't need to do anything for the `down`, which also seems KISS! What do you think?

Reply:
> I would suggest that because this migration is purely to correct the state of the database that it doesn't need to do anything for the down

+1 on this, don't think we should be adding `down` for these migrations

Reply:
Does the down migration for the original addition of groups set the type back? That’s what I was thinking of but it should be in the OG migration not individual ones.

So convention for new settings is no down.

But I think we may have problems with rollbacks for new settings if they go past the addition of the column

Reply:
This says (and looks to me) that it does! https://github.com/TryGhost/Ghost/blob/master/core/server/data/migrations/versions/3.22/04-populate-settings-groups-and-flags.js#L169-L192

> So convention for new settings is no down.

:+1: Sounds good

> But I think we may have problems with rollbacks for new settings if they go past the addition of the column

Yeah maybe, we could try an split migrations into 1) make sure the group is correct and 2) anything else you need?

Reply:
The [down](https://github.com/TryGhost/Ghost/blob/master/core/server/data/migrations/versions/3.22/04-populate-settings-groups-and-flags.js#L169-L191) for original migration to populate groups does set the `type` back to original values, and doesn't reset the group. 

So yeah you are right its gonna be a mixed mess in settings DB when we rollback to before 3.22 after adding new settings as the OG settings will have a proper type and the new ones will be still in DB with `type` as `string` etc. 😬 Not sure of the side-effects in that case if it breaks something or if they'll be just ignored, ideally they should not even be in DB at that point.